Results exports

jeffkinnison opened the pull request:
Added functions to export searches to CSV (with user-determined delimiter), XML, and JSON to `tesserae.utils.export`. Also added a CLI script to call this functionality, accessible through `tesserae.cli.export`.

jeffkinnison commented on tesserae/cli/export.py:
```diff
@@ -0,0 +1,99 @@
+        args = args + (delimiter,)
+
+
+    if filepath:
+        dump(*args)
+    else:
+        print(dumps(*args))
```
Reworking filetype routing to be in tesserae/utils/exports/__init__.py and handled automatically. This was partially to avoid unnecessary imports and partially to route everything to single function calls.

Since there isn't any persistent data, a class interface seems like overkill. Honestly, if there was one thing I could undo in this code base, it would be the tendency to define classes instead of just using functions.

jeffkinnison commented on tesserae/cli/export.py:
```diff
@@ -0,0 +1,99 @@
+def main(connection, search_id, format, filepath=None, delimiter=','):
+    if format == 'json':
+        from tesserae.utils.exports.json import dump, dumps
+        args = (connection, search_id)
+        if filepath:
+            args = args + (filepath,)
```
Reworking filetype routing to be in tesserae/utils/exports/__init__.py and handled automatically. This was partially to avoid unnecessary imports and partially to route everything to single function calls.

jeffkinnison commented:
Reworked things based on the comments. I'll get to the unit tests at some point, need to be able to see the mini vergil/lucan results to do that.

jeffkinnison commented on tesserae/utils/exports/__init__.py:
```diff
@@ -0,0 +1,197 @@
+    ----------
+    connection : tesserae.db.TessMongoConnection
+        Connection to the MongoDB instance.
+    search_id : str or `bson.objectid.ObjectID`
```
I think the TessMongoConnection should work with either string or ObjectID, no?

nOkuda replied:
My comment suggests I didn't think this was the case. We could make letting either str or ObjectID be accepted as a separate issue.

jeffkinnison commented:
Changes:
- Fixed potential issues in Paging object and added documentation
- Standardized `search_id` to `bson.objectid.ObjectID`
- Updated variable names for consistency in general

nOkuda commented:
Thank you for your recent changes. There's still the `args.delimiter` note that I think should be addressed before I approve for merging.

jeffkinnison commented:
It should have been fixed before, and should have been included in the update list I posted.
